modules/gg-fleet-statusd/src/entry.c

```diff
@@ -104,14 +114,79 @@ static void connection_status_close_callback(void *ctx, uint32_t handle) {
     // TODO: Add reconnects (on another thread or with timer
 }
 
+static bool is_positive_integer(GglBuffer str) {
+    if (str.len == 0) {
```

Comment:
It seems possible that we could pass a string that isn't null-terminated to the C library function below 
```suggestion
    if ((str.len == 0) || (str.len > 19)) {
```